src/main/java/com/google/vinet/data/MatchingRunner.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.data;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.*;
+
+public class MatchingRunner {
+  private DatastoreService datastore;
+  private Set<IsolateTimeSlot> isolateTimeSlots;
+  private Set<VolunteerTimeSlot> volunteerTimeSlots;
+  private LocalDate date;
+
+  /**
+   * Construct a MatchingRunner with none of its data set.
+   */
+  public MatchingRunner() {
+
+  }
```

Comment:
What's the purpose of an empty constructor?

Reply:
Just to clearly communicate that the class can be instantiated with none of its data set at construction time. If it was omitted, though it would still be possible to call `new MatchingRunner()`, it wouldn't be clear if this was an intended course of action.